src/hub/ShareClassManager.sol

```diff
@@ -312,6 +323,58 @@ contract ShareClassManager is Auth, IShareClassManager {
         );
     }
 
+    /// @inheritdoc IShareClassManager
+    function updatePricePerShare(PoolId poolId, ShareClassId scId_, D18 navPoolPerShare) external auth {
+        require(exists(poolId, scId_), ShareClassNotFound());
+
+        ShareClassMetrics storage m = metrics[scId_];
+        m.navPerShare = navPoolPerShare;
+        emit UpdateShareClass(
+            poolId, scId_, navPoolPerShare.mulUint128(m.totalIssuance), navPoolPerShare, m.totalIssuance
+        );
+    }
+
+    /// @inheritdoc IShareClassManager
+    function updateMetadata(
+        PoolId poolId,
+        ShareClassId scId_,
+        string calldata name,
+        string calldata symbol,
+        bytes32 salt,
+        bytes calldata
+    ) external auth {
+        require(exists(poolId, scId_), ShareClassNotFound());
+
+        _updateMetadata(scId_, name, symbol, salt);
+
+        emit UpdateMetadata(poolId, scId_, name, symbol, salt);
+    }
```

Comment:
NIT

Can we also expose that in the hub? We wanted that, right?

Reply:
Yeah, its part of the `Increase Hub interface` ticket: https://www.notion.so/Feat-Increase-Hub-interface-1cb2eac24e17801f9f60d673927ef4eb?pvs=4#1d62eac24e178085a267c7ff2f54809e

Reply:
I'm adding it here now actually! Also simplifying the salt logic a bit then